transformer_engine/pytorch/module/layernorm_linear.py

```diff
@@ -473,14 +473,20 @@ def backward(
             if not ctx.use_bias:
                 grad_bias = None
 
-        # Handle custom DDP from mcore.
-        weight.grad_added_to_main_grad = ctx.fuse_wgrad_accumulation
+        if weight.requires_grad:
+            # Handle custom DDP from mcore.
```

Comment:
Maybe create a smaller helper method that takes in `weight` and `fuse_wgrad_accumulation` arguments? Then you don't need these nested if conditions everywhere.